framework/src/samplers/Sampler.C

```diff
@@ -12,103 +12,76 @@
 /*            See COPYRIGHT for full restrictions               */
 /****************************************************************/
 
+// MOOSE includes
 #include "Sampler.h"
+#include "MooseRandom.h"
 #include "Distribution.h"
-#include <limits>
 
 template <>
 InputParameters
 validParams<Sampler>()
 {
   InputParameters params = validParams<MooseObject>();
-  params += validParams<RandomInterface>();
-  params.addParam<bool>("reseed", false, "Reseed for each new sample if this value is true");
-  params.addRequiredParam<std::vector<std::string>>(
-      "perturb_parameters", "The names of the parameters that you want to perturb");
+  params += validParams<SetupInterface>();
+  params += validParams<DistributionInterface>();
+
+  params.addClassDescription("A base class for distribution sampling.");
+  params.set<std::vector<OutputName>>("outputs") = {"none"};
   params.addRequiredParam<std::vector<DistributionName>>(
-      "distributions",
-      "The names of distributions that you want to use to perturb the given parameters");
+      "distributions", "The names of distributions that you want to sample.");
+  params.addParam<std::vector<unsigned int>>("seeds",
+                                             std::vector<unsigned int>(1, 19800624),
```

Comment:
It is just a number I selected, we can change it (it's my birthday).

We can try to improve the seeds, but I didn't want to limit the number of seeds available to the user. Take a look at the SOBOL sampler, which has two independent matrices that need to be create, it would be nice to be able to control the seeds for these separately.